tools/topology/topology2/include/pipelines/cavs/eqfir-capture.conf

```diff
@@ -0,0 +1,152 @@
+#
+# CAVS fir eq capture pipeline
+#
+# A simple fir eq pipeline. All attributes defined herein are namespaced
+# by alsatplg to "Object.Pipeline.eqfir-capture.N.attribute_name".
+#
+# Usage: eqfir-capture pipeline object can be instantiated as:
+#
+# Object.Pipeline.eqfir-capture."N" {
+# 	format		"s16le"
+# 	period		1000
+# 	time_domain	"timer"
+# 	channels	2
+# 	rate		48000
+# }
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+<include/common/audio_format.conf>
+<include/components/copier.conf>
+<include/components/gain.conf>
+<include/components/pipeline.conf>
+<include/components/eqfir.conf>
+<include/controls/bytes.conf>
+<include/common/data.conf>
+
+Class.Pipeline."eqfir-capture" {
+
+	DefineAttribute."index" {}
+
+	<include/pipelines/pipeline-common.conf>
+
+	attributes {
+		!constructor [
+			"index"
+		]
+
+		!immutable [
+			"direction"
+		]
+
+		#
+		# gain-capture objects instantiated within the same alsaconf node must have
+		# unique pipeline_id attribute
+		#
+		unique	"instance"
+	}
+
+	Object.Widget {
+		copier."1" {
+			copier_type	"host"
+			type	"aif_out"
+			num_audio_formats 3
+			# 16-bit 48KHz 2ch
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			# 24-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	24
+				dma_buffer_size "$[$obs * 2]"
+			}
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.3 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			node_type $HDA_HOST_INPUT_CLASS
+		}
+
+		eqfir."1" {
+			num_audio_formats 2
+			#16-bit 48KHz 2ch
+			Object.Base.audio_format.1 {}
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+			}
+
+			Object.Control.bytes."1" {
+				Object.Base.data."fir_eq" {
+					bytes "
```

Comment:
@juimonen I assume this is the coefficients ? This should really be passed by file reference now e.g.
```
file $EQFIR1_COEFFS
```
The EQFIR coefficient file would be generated by Octave scripts from @singalsu

Reply:
Yep, this need to be cleaned up and those tplg2 and ipc4 style versions generated. We are now debugging the ipc4 blobs system with these topologies.